scripts/source_files/helper_functions.sh

```diff
@@ -1,34 +1,28 @@
 #!/usr/bin/env bash
 
-#Update the version property in a gradle properties file
-function update_version(){
-  [ $# -eq 2 ] || return 11
+function __update_a_version_property(){
+  [ $# -eq 3 ] || return 11
   local target_version="$1"
   local target_file="$2"
-  local version_property="version"
-  local version_matcher="${version_property}=.*"
+  local version_property="$3"
+  local version_matcher="${version_property} ?= ?.*"
```

Comment:
For reference, what does ?= do here?

Reply:
that is a regex expression. so in the current version we have ( ?= ?). the parentheses denote a capturing group that allows us to record the contents and drop them into our replacement expression later. 
The ? is an "optional" signifier, denoting that the character (whitespace in our case) need not be there.

Reply:
Had to get rid of the regex, regex are evaluated differently on MacOS (for our android tests) and Unix (all the rest). it might have been possible but it would have been confusing at least.